Core/HLE/sceUsbCam.cpp

```diff
@@ -50,6 +52,17 @@ static int sceUsbCamStopMic() {
 	return 0;
 }
 
+static int sceUsbCamReadMicBlocking(u32 bufAddr, u32 size) {
+	INFO_LOG(HLE, "UNIMPL sceUsbCamReadMicBlocking: size: %d", size);
+	for (unsigned int i = 0; i < size; i++) {
+		if (Memory::IsValidAddress(bufAddr + i)) {
+			Memory::Write_U8(std::rand() & 0xFF, bufAddr + i);
+		}
+	}
+	std::this_thread::sleep_for(std::chrono::milliseconds(1000 / micParam.frequency * (size / 2)));
```

Comment:
No, this needs to block the HLE thread, not the actual C++ thread.  Look at sceAudio's blocking functions.

Sleeping the CPU thread will just pause emulation for that many milliseconds.  We don't emulate PSP threads as real C++ threads for multiple reasons, including:
 * Thread scheduling has very specific behavior in PSP's OS, and deviating causes games to break.
 * The PSP is single core, and games crash and glitch when we run two PSP threads concurrently on multi-core systems (because the PSP game's code was not written correctly for multithreading and accidentally relies on single core behavior.)
 * Games sleep PSP threads for sometimes specific portions of a millisecond, but many modern operating systems (for example, Windows) are incapable of sleeping threads for less than a full millisecond.

-[Unknown]